pkg/deploy/dashboard/dashboard.go

```diff
@@ -33,97 +34,98 @@ var (
 	log = ctrl.Log.WithName("dashboard")
 )
 
-type Dashboard struct {
-	deployContext *deploy.DeployContext
-	component     string
+type DashboardReconciler struct {
+	deploy.Reconcilable
 }
 
-func NewDashboard(deployContext *deploy.DeployContext) *Dashboard {
-	return &Dashboard{
-		deployContext: deployContext,
-		component:     deploy.DefaultCheFlavor(deployContext.CheCluster) + "-dashboard",
-	}
+func NewDashboardReconciler() *DashboardReconciler {
+	return &DashboardReconciler{}
 }
 
-func (d *Dashboard) GetComponentName() string {
-	return d.component
+func (d *DashboardReconciler) getComponentName(ctx *deploy.DeployContext) string {
+	return deploy.DefaultCheFlavor(ctx.CheCluster) + "-dashboard"
 }
 
-func (d *Dashboard) Reconcile() (done bool, err error) {
+func (d *DashboardReconciler) Reconcile(ctx *deploy.DeployContext) (reconcile.Result, bool, error) {
 	// Create a new dashboard service
-	done, err = deploy.SyncServiceToCluster(d.deployContext, d.component, []string{"http"}, []int32{8080}, d.component)
+	done, err := deploy.SyncServiceToCluster(ctx, d.getComponentName(ctx), []string{"http"}, []int32{8080}, d.getComponentName(ctx))
 	if !done {
-		return false, err
+		return reconcile.Result{}, false, err
```

Comment:
Do we need reconcile.Result{Requeue: true} here ?

Reply:
No. In general we need Requeue if we deal with non cached object (cluster ones).